lib/MusicBrainz/Server/Controller/Artist.pm

```diff
@@ -283,6 +280,29 @@ sub show : PathPart('') Chained('load')
     push(@identities, @other_identities);
     $c->stash(other_identities => \@other_identities,
               identities => \@identities);
+
+    $c->stash(
+        current_view => 'Node',
+        component_path => 'artist/ArtistIndex',
+        component_props => {
+            ajaxFilterFormUrl => $c->uri_for_action('/ajax/filter_artist_release_groups_form', { artist_id => $artist->id }),
+            artist => $artist,
+            filterForm => $c->stash->{filter_form},
+            hasFilter => boolean_to_json($c->stash->{has_filter}),
+            includingAll => $including_all,
+            includingVarious => $including_va,
+            legalName => $legal_name,
```

Comment:
$legal_name used to be stashed in https://github.com/metabrainz/musicbrainz-server/blob/9a996d78a40374c77127a303786224b18e2fc538/lib/MusicBrainz/Server/Controller/Artist.pm#L249 way before https://github.com/metabrainz/musicbrainz-server/blob/9a996d78a40374c77127a303786224b18e2fc538/lib/MusicBrainz/Server/Controller/Artist.pm#L273

This though is passed *after* that line, so it will always pass (and show) the artist itself if there's no legal name artist.

Reply:
Oh right, thanks. That's confusing - the `$artist` isn't necessarily a legal name (?) and it seems to be only needed for the `$legal_name->relationships` call below to get `@other_identities`. So I think I'd prefer removing this assignment and doing `($legal_name // $artist)->relationships` below, rather than clobbering the variable.

Reply:
Sounds like a good plan to me.